### PR TITLE
Show max 20 canvases

### PR DESCRIPTION
**Pull request by mkamonMdt**
The PR provides implementation for #39 
The final output of above commits:
1 - On old view I added "Open Peers" tab that opens the PeerList 
2 - The PeerList view 
3- Filtering functionality 

Screenshot from 2024-06-05 10-29-17

### CODE REVIEW

**Comment by darioalessandro** [comment]
this is awesome @mkamonMdt !!! testing now

**Comment by darioalessandro** [comment]
@mkamonMdt I think clippy is angry:

```
error: this expression creates a reference which is immediately dereferenced by the compiler
  --> videocall-client/src/decode/hash_map_with_ordered_keys.rs:93:29
   |
93 |             self.map.remove(&key);
   |                             ^^^^ help: change this to: `key`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
   = note: `-D clippy::needless-borrow` implied by `-D warnings`

error: unneeded `return` statement
   --> videocall-client/src/decode/peer_decode_manager.rs:168:9
    |
168 |         return false;
    |         ^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
    = note: `-D clippy::needless-return` implied by `-D warnings`
help: remove `return`
    |
168 -         return false;
168 +         false
    |

error: could not compile `videocall-client` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
make: *** [Makefile:24: check] Error 101
```

File: yew-ui/src/components/peer_list.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                <ul>
+                    { for filtered_peers.iter().map(|peer| 
+                        html!{
+                            <li><PeerListItem name={peer.clone()}/></li>
+                        })
+                    }
+                </ul>
+            </div>
+            <div class="footer">
+                <p>{ "Some foot note" }</p>
```
Comment (darioalessandro) [inline yew-ui/src/components/peer_list.rs:79]: I think we should remove this

**Comment by mkamonMdt** [comment]
> fixed linter on main, I would just remove the footer placeholder and call it a day

Thanks for fixing it.

**Pull request merged**